### PR TITLE
planner: simplify the global binding interface

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #48875

Problem Summary: planner: simplify the global binding interface

### What changed and how does it work?

1. Rename `Update` to `LoadFromStorageToCache`;
2. Remove `LockBindInfoSQL` and `ReloadGlobalBindings`;
3. Update global binding cache through `LoadFromStorageToCache` only

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by AilinKid — APPROVED** [review]

File: pkg/session/bootstrap_test.go [hunk 1]
```diff
@@ -526,74 +526,6 @@ type bindTestStruct struct {
 	deleteText   string
 }
 
-func TestUpdateBindInfo(t *testing.T) {
```
Comment (qw4990) [inline pkg/session/bootstrap_test.go:529]: This test is too old

**Review by hawkingrei — APPROVED** [review]

**Review by zimulala — APPROVED** [review]

**Comment by hawkingrei** [comment]
/unhold

schedule CI by hand.

**Pull request merged**